core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -181,6 +182,7 @@ class GroupCoordinator(val brokerId: Int,
           responseCallback(JoinGroupResult(memberId, Errors.UNKNOWN_MEMBER_ID))
         case Some(group) =>
           group.inLock {
+            val joinReason = reason.getOrElse("unknown reason")
```

Comment:
If we do this, it might be better to not use an `Option` after all. We could simply provided the default reason to `handleJoinGroup` if none is provided. Also, how about using `not provided` instead of `unknown reason`?

Reply:
good idea! i'll update the leave group as well.